zkvm/tests/zkvm.rs

```diff
@@ -79,3 +89,202 @@ fn issue() {
         }
     };
 }
+
+// questions:
+// - do we have to use separate nonce predicates for the two inputs (to ensure uniqueness if they're same qty?)
+// - is it enough (for testing) to get the value from the issue contract? (how do we create a "utxo" in one tx?)
+fn spend_2_2_contract(
+    input_1: u64,
+    input_2: u64,
+    output_1: u64,
+    output_2: u64,
+    flv: Scalar,
+    issuance_pred: &Predicate,
+    nonce_pred: &Predicate,
+    recipient_1_pred: &Predicate,
+    recipient_2_pred: &Predicate,
+) -> Vec<Instruction> {
+    let mut instructions = vec![];
+    instructions.append(&mut issue_helper(input_1, flv, issuance_pred, nonce_pred)); // stack: issued-value-1
+    instructions.append(&mut issue_helper(input_2, flv, issuance_pred, nonce_pred)); // stack: issued-value-1, issued-value-2
+
+    instructions.push(Instruction::Push(
+        Commitment::from(CommitmentWitness::blinded(output_1)).into(),
+    )); // stack: issued-value-1, issued-value-2, output-1-quantity
+    instructions.push(Instruction::Push(
+        Commitment::from(CommitmentWitness::blinded(flv)).into(),
+    )); // stack: issued-value-1, issued-value-2, output-1-quantity, output-1-flavor
+
+    instructions.push(Instruction::Push(
+        Commitment::from(CommitmentWitness::blinded(output_2)).into(),
+    )); // stack: ... output-2-quantity
+    instructions.push(Instruction::Push(
+        Commitment::from(CommitmentWitness::blinded(flv)).into(),
+    )); // stack: ... output-2-quantity, output-2-flavor
+
+    instructions.push(Instruction::Cloak(2, 2)); // stack: output-1, output-2
+
+    instructions.push(Instruction::Push(recipient_2_pred.clone().into())); // stack: issued-value-1, issued-value-2, recipient-2-pred
+    instructions.push(Instruction::Output(1)); // stack: issued-value-1
+    instructions.push(Instruction::Push(recipient_1_pred.clone().into())); // stack: issued-value-1, recipient-1-pred
+    instructions.push(Instruction::Output(1)); // stack: empty
+
+    instructions
+}
+
+#[test]
+fn spend_2_2() {
+    let (tx, _txid, txlog) = {
+        // Generate predicates
+        let issuance_pred =
+            Predicate::from_witness(PredicateWitness::Key(Scalar::from(0u64))).unwrap();
+        let nonce_pred =
+            Predicate::from_witness(PredicateWitness::Key(Scalar::from(1u64))).unwrap();
+        let recipient_1_pred =
+            Predicate::from_witness(PredicateWitness::Key(Scalar::from(2u64))).unwrap();
+        let recipient_2_pred =
+            Predicate::from_witness(PredicateWitness::Key(Scalar::from(3u64))).unwrap();
+
+        // Generate flavor scalar
+        let mut t = Transcript::new(b"ZkVM.issue");
+        t.commit_bytes(b"predicate", issuance_pred.point().as_bytes());
+        let flavor = t.challenge_scalar(b"flavor");
+
+        // Build program
+        let program = spend_2_2_contract(
```

Comment:
I think if we were to call issue_helper here in the test, and then build that instruction & pass the input from txlog into spend_2_2_contract that might be a good way to simulate spending UTXOs?